performance_test/src/experiment_execution/analysis_result.hpp

```diff
@@ -17,20 +17,76 @@
 
 #include <sys/time.h>
 #include <sys/resource.h>
-
+#include <boost/uuid/uuid_io.hpp>
 #include <chrono>
 #include <sstream>
 #include <string>
 
 #include "../utilities/statistics_tracker.hpp"
+#ifdef ODB_FOR_SQL_ENABLED
+  #include <odb/core.hxx>
+  #include "experiment_configuration.hpp"
+#endif
 
 namespace performance_test
 {
 
 /// Outstream operator for timeval to seconds (double).
 std::ostream & operator<<(std::ostream & stream, const timeval & e);
+#ifdef ODB_FOR_SQL_ENABLED
+class RusageTracker
+{
+public:
+  void setValues(rusage sys_usage) const
+  {
+    m_ru_utime = sys_usage.ru_utime;
+    m_ru_stime = sys_usage.ru_stime;
+    m_ru_maxrss = sys_usage.ru_maxrss;
+    m_ru_ixrss = sys_usage.ru_ixrss;
+    m_ru_idrss = sys_usage.ru_idrss;
+    m_ru_isrss = sys_usage.ru_isrss;
+    m_ru_minflt = sys_usage.ru_minflt;
+    m_ru_majflt = sys_usage.ru_majflt;
+    m_ru_nswap = sys_usage.ru_nswap;
+    m_ru_inblock = sys_usage.ru_inblock;
+    m_ru_oublock = sys_usage.ru_oublock;
+    m_ru_msgsnd = sys_usage.ru_msgsnd;
+    m_ru_msgrcv = sys_usage.ru_msgrcv;
+    m_ru_nsignals = sys_usage.ru_nsignals;
+    m_ru_nvcsw = sys_usage.ru_nvcsw;
+    m_ru_nivcsw = sys_usage.ru_nivcsw;
+  }
+
+private:
+  friend class odb::access;
+  mutable struct timeval m_ru_utime = {};
```

Comment:
Store here nanoseconds for consistency in the DB.

Reply:
done